src/esp/gfx/CubeMapCamera.cpp

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree
+
+#include "CubeMapCamera.h"
+#include <Magnum/EigenIntegration/Integration.h>
+
+namespace Mn = Magnum;
+namespace Cr = Corrade;
+
+namespace esp {
+namespace gfx {
+
+CubeMapCamera::CubeMapCamera(scene::SceneNode& node) : RenderCamera(node) {
+  updateOriginalViewingMatrix();
+}
+CubeMapCamera::CubeMapCamera(scene::SceneNode& node,
+                             const vec3f& eye,
+                             const vec3f& target,
+                             const vec3f& up)
+    : CubeMapCamera(node,
+                    Mn::Vector3{eye},
+                    Mn::Vector3{target},
+                    Mn::Vector3{up}) {}
+CubeMapCamera::CubeMapCamera(scene::SceneNode& node,
+                             const Mn::Vector3& eye,
+                             const Mn::Vector3& target,
+                             const Mn::Vector3& up)
+    : RenderCamera(node, eye, target, up) {
+  updateOriginalViewingMatrix();
+}
+CubeMapCamera& CubeMapCamera::updateOriginalViewingMatrix() {
+  originalViewingMatrix_ = this->node().transformation();
+  return *this;
+}
+
+CubeMapCamera& CubeMapCamera::restoreTransformation() {
+  this->node().setTransformation(originalViewingMatrix_);
+  return *this;
+}
+
+CubeMapCamera& CubeMapCamera::switchToFace(unsigned int cubeSideIndex) {
+  switch (cubeSideIndex) {
+    case 0:
+      switchToFace(Mn::GL::CubeMapCoordinate::PositiveX);
+      break;
+    case 1:
+      switchToFace(Mn::GL::CubeMapCoordinate::NegativeX);
+      break;
+    case 2:
+      switchToFace(Mn::GL::CubeMapCoordinate::PositiveY);
+      break;
+    case 3:
+      switchToFace(Mn::GL::CubeMapCoordinate::NegativeY);
+      break;
+    case 4:
+      switchToFace(Mn::GL::CubeMapCoordinate::PositiveZ);
+      break;
+    case 5:
+      switchToFace(Mn::GL::CubeMapCoordinate::NegativeZ);
+      break;
```

Comment:
FYI the enums are defined in GL to have consecutive indices, so you can do something like 

```cpp
switchToFace(Mn::GL::CubeMapCoordinate(int(Mn::GL::CubeMapCoordinate::PositiveX) + cubeSideIndex));
```

instead of the giant switch.

Reply:
Convenient!